uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/impl/KeyboardOperationInvokeContextMenuForSelectedCell.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.uberfire.ext.wires.core.grids.client.widget.grid.impl;
+
+import com.google.gwt.event.dom.client.KeyCodes;
+import org.uberfire.ext.wires.core.grids.client.model.GridData;
+import org.uberfire.ext.wires.core.grids.client.util.ColumnIndexUtilities;
+import org.uberfire.ext.wires.core.grids.client.widget.grid.GridWidget;
+import org.uberfire.ext.wires.core.grids.client.widget.layer.GridLayer;
+
+public class KeyboardOperationInvokeContextMenuForSelectedCell extends BaseKeyboardOperation {
+
+    public KeyboardOperationInvokeContextMenuForSelectedCell(final GridLayer gridLayer) {
+        super(gridLayer);
+    }
+
+    @Override
+    public TriStateBoolean isControlKeyDown() {
```

Comment:
@jomarko @manstis 
please avoid such kind of things like "TriStateBoolean" at all. I did not see it before but looking at it is plainly a huge source of (unneeded) confusion.

Reply:
I disagree; some keyboard operations may care if Control is pressed or if it is not explicitly pressed or not care either way.

Reply:
@manstis 
I think it is responsibility of the final implementation to decide
1) if it is needed to check the "ctrl" pressed status
2) in case, if the Ctrl is actually pressed or not

Collapsing this two decision in a single one in this way I think lead only to confusion